src/Test/Utilities/Shared/Compilation/OperationTreeVerifier.cs

```diff
@@ -0,0 +1,1046 @@
+        {
+            var valueStr = constant != null ? constant.ToString() : "null";
+            LogString($"{header}: {valueStr}");
+        }
+
+        private void LogSymbol(ISymbol symbol, string header, bool logDisplayString = true)
+        {
+            if (!string.IsNullOrEmpty(header))
+            {
+                LogString($"{header}: ");
+            }
+
+            var symbolStr = symbol != null ? (logDisplayString ? symbol.ToTestDisplayString() : symbol.Name) : "null";
+            LogString($"{symbolStr}");
+        }
+
+        private void LogType(ITypeSymbol type)
+        {
+            var typeStr = type != null ? type.ToTestDisplayString() : "null";
+            LogString($"Type: {typeStr}");
+        }
+
+        #endregion
+
+        #region Visit methods
+
+        public override void Visit(IOperation operation)
+        {
+            if (operation == null)
+            {
+                return;
+            }
+
+            if (operation != _root)
+            {
+                Indent();
+            }
+
+            base.Visit(operation);
+
+            if (operation != _root)
+            {
+                Unindent();
+            }
+        }
+
+        private void Visit(IOperation operation, string header)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(header));
+
+            if (operation == null)
+            {
+                return;
+            }
+
+            Indent();
+            LogString($"{header}: ");
+            Visit(operation);
+            Unindent();
+        }
+
+        private void VisitArray<T>(ImmutableArray<T> list, string header)
+            where T : IOperation
+        {
+            Debug.Assert(!string.IsNullOrEmpty(header));
+
+            if (list.IsDefaultOrEmpty)
+            {
+                return;
+            }
+
+            Indent();
+            LogString($"{header}: ");
+            VisitArray(list);
+            Unindent();
+        }
+
+        private void VisitInstanceExpression(IOperation instance)
+        {
+            Visit(instance, header: "Instance Receiver");
+        }
+
+        internal override void VisitNoneOperation(IOperation operation)
+        {
+            Assert.True(false, "Encountered an IOperation with `Kind == OperationKind.None` while walking the operation tree.");
+        }
+
+        public override void VisitBlockStatement(IBlockStatement operation)
+        {
+            LogString(nameof(IBlockStatement));
+
+            var statementsStr = $"{operation.Statements.Length} statements";
+            var localStr = !operation.Locals.IsEmpty ? $", {operation.Locals.Length} locals" : string.Empty;
+            LogString($" ({statementsStr}{localStr})");
+            LogCommonPropertiesAndNewLine(operation);
+
+            if (operation.Statements.IsEmpty)
+            {
+                return;
+            }
+
+            LogLocals(operation.Locals);
+
+            base.VisitBlockStatement(operation);
```

Comment:
@genlu Moved to using base visitors where we don't do custom logging for child nodes.